packages/apptive_grid_heinzelmen/lib/src/invitation_dialog/invitation_dialog.dart

```diff
@@ -0,0 +1,344 @@
+/// This will return a Future<bool> which will return if one or more invites where send
+Future<bool> showSpaceInvitationDialog({
+  required BuildContext context,
+  required Space space,
+  List<Role> allowedRoles = Role.values,
+  ApptiveLinkType type = ApptiveLinkType.invite,
+  InvitationLocalization? localization,
+}) async {
+  assert(type == ApptiveLinkType.invite || type == ApptiveLinkType.addShare);
+  assert(space.links.containsKey(type));
+  assert(allowedRoles.isNotEmpty);
+
+  final l10n = localization ??
+      (Localizations.localeOf(context).languageCode == 'de'
+          ? const InvitationLocalizationDE()
+          : const InvitationLocalizationEn());
+
+  TextEditingController emailController = TextEditingController();
+  ValueNotifier<Role> roleNotifier = ValueNotifier(allowedRoles.first);
+  ValueNotifier<bool> loading = ValueNotifier(false);
+  ValueNotifier<dynamic> errorNotifier = ValueNotifier(null);
+  bool dialogResult = false;
+  return showDialog(
+    context: context,
+    barrierDismissible: false,
+    builder: (dialogContext) {
+      return ValueListenableBuilder<bool>(
+        valueListenable: loading,
+        builder: (_, loading, child) => IgnorePointer(
+          ignoring: loading,
+          child: child,
+        ),
+        child: Form(
+          child: AlertDialog(
+            title: Text.rich(
+              TextSpan(
+                children: getSpans(
+                  getText: l10n.title,
+                  argument: space.name,
+                  style: TextStyle(
+                    color: Theme.of(context).colorScheme.primary,
+                  ),
+                ),
+              ),
+            ),
+            content: _InvitationDialogContent(
+              emailController: emailController,
+              role: roleNotifier,
+              error: errorNotifier,
+              allowedRoles: allowedRoles,
+              localization: l10n,
+            ),
+            actions: [
+              TextButton(
+                onPressed: Navigator.of(dialogContext).pop,
+                child: Text(l10n.actionCancel),
+              ),
+              _InvitationSendButton(
+                loading: loading,
+                role: roleNotifier,
+                error: errorNotifier,
+                onInviteSend: (value) => dialogResult = value,
+                localization: l10n,
+                email: emailController,
+                link: space.links[type]!,
+              ),
+            ],
+          ),
+        ),
+      );
+    },
+  ).then((_) => dialogResult);
+}
+
+/// Returns a List of [TextSpans] that format [argument] with [style] and use default styles for other text that was generated with [getText]
+@visibleForTesting
+List<TextSpan> getSpans({
+  required String Function(String) getText,
+  required String argument,
+  required TextStyle style,
```

Comment:
Could we rename the arguments and method to be more understandable? Maybe something like `getHighlighted(getText, highlight, highlightStyle)`.

Reply:
I have adjusted the function and added some more explanation to the documentation